core/src/main/java/org/mskcc/cbio/portal/scripts/ImportResourceDefinition.java

```diff
@@ -0,0 +1,330 @@
+        // if bulkLoading is ever turned off,
+        // code has to be added to check whether
+        // a resource data update should be
+        // perform instead of an insert
+        MySQLbulkLoader.bulkLoadOn();
+
+        if (relaxed) {
+            MySQLbulkLoader.relaxedModeOn();
+        }
+
+        FileReader reader = new FileReader(resourceDataFile);
+        BufferedReader buff = new BufferedReader(reader);
+
+        String line = buff.readLine();
+        String[] headerNames = splitFields(line);
+
+        // validate columns and get index
+        int resourceIdIndex = findAndValidateResourceIdColumn(headerNames);
+        int displayNameIndex = findAndValidateDisplayNameColumn(headerNames);
+        int descriptionIndex = findAndValidateDescriptionColumn(headerNames);
+        int resourceTypeIndex = findAndValidateResourceTypeColumn(headerNames);
+        int openByDefaultIndex = findAndValidateOpenByDefaultColumn(headerNames);
+        int priorityIndex = findAndValidatePriorityColumn(headerNames);
+
+        while ((line = buff.readLine()) != null) {
+            if (skipLine(line.trim())) {
+                continue;
+            }
+
+            String[] fieldValues = getFieldValues(line, headerNames);
+
+            // set default value
+            String resourceId = "";
+            String displayName = "";
+            String description = "";
+            String resourceType = "";
+            Boolean openByDefault = false;
+            int priority = 1;
+            // get resource definitions from columns
+            // get resourceId
+            if (resourceIdIndex != -1 && isValueNotMissing(fieldValues[resourceIdIndex].toUpperCase())) {
+                resourceId = fieldValues[resourceIdIndex].toUpperCase();
+            } else {
+                throw new RuntimeException(
+                        "Please provide a valid resource id");
+            }
+            // get displayName
+            if (resourceIdIndex != -1 && isValueNotMissing(fieldValues[displayNameIndex])) {
+                displayName = fieldValues[displayNameIndex];
+            }
+            else {
+                throw new RuntimeException(
+                        "Please provide a valid display name");
+            }
+            // get description (optional)
+            if (descriptionIndex != -1 && isValueNotMissing(fieldValues[descriptionIndex])) {
+                description = fieldValues[descriptionIndex];
+            }
+            // get resourceType (must be value of ResourceTypes)
+            if (resourceTypeIndex != -1 && isValidResourceType(fieldValues[resourceTypeIndex])) {
+                resourceType = fieldValues[resourceTypeIndex];
+            }
+            // get openByDefault (optional)
+            if (openByDefaultIndex != -1 && isValidOpenByDefault(fieldValues[openByDefaultIndex])) {
+                openByDefault = fieldValues[openByDefaultIndex].equalsIgnoreCase("true");
```

Comment:
here you just want to parse the boolean value right? and if it's not "true" or "false" then the default would be false?

the `Boolean` java docs describes a built in method for parsing a boolean from a string and I think it's the exact behavior you're looking for too

https://docs.oracle.com/javase/8/docs/api/java/lang/Boolean.html

Reply:
Thanks! Good to know this!